.azure/test-linux.yml

```diff
@@ -0,0 +1,140 @@
+parameters:
+  - name: "pythonVersion"
+    type: string
+    displayName: "Version of Python to test"
+
+  - name: "testQPY"
+    type: boolean
+    default: false
+
+  - name: "testImages"
+    type: boolean
+    default: false
+
+
+jobs:
+  - job: "Linux_Tests_Python${{ replace(parameters.pythonVersion, '.', '') }}"
+    displayName: "Test Linux Python ${{ parameters.pythonVersion }}"
+    pool: {vmImage: 'ubuntu-latest'}
+
+    variables:
+      QISKIT_SUPPRESS_PACKAGING_WARNINGS: Y
+      PIP_CACHE_DIR: $(Pipeline.Workspace)/.pip
+      QISKIT_TEST_CAPTURE_STREAMS: 1
+
+    steps:
+      - task: UsePythonVersion@0
+        inputs:
+          versionSpec: '${{ parameters.pythonVersion }}'
+        displayName: 'Use Python ${{ parameters.pythonVersion }}'
+
+      - task: Cache@2
+        inputs:
+          key: 'pip | "$(Agent.OS)" | "${{ parameters.pythonVersion }}" |"$(Build.BuildNumber)"'
+          restoreKeys: |
+            pip | "$(Agent.OS)" | "${{ parameters.pythonVersion }}"
+            pip | "$(Agent.OS)"
+            pip
+          path: $(PIP_CACHE_DIR)
+        displayName: "Cache pip"
+
+      - task: Cache@2
+        inputs:
+          key: 'stestr | "$(Agent.OS)" | "${{ parameters.pythonVersion }}" | "$(Build.BuildNumber)"'
+          restoreKeys: |
+            stestr | "$(Agent.OS)" | "${{ parameters.pythonVersion }}"
+            stestr | "$(Agent.OS)"
+            stestr
+          path: .stestr
+        displayName: "Cache stestr"
+
+      - bash: |
+          set -e
+          python -m pip install --upgrade pip setuptools wheel virtualenv
+          virtualenv test-job
+          source test-job/bin/activate
+          pip install -U -r requirements.txt -r requirements-dev.txt -c constraints.txt
+          python setup.py sdist
+          pip install -U -c constraints.txt dist/qiskit-terra*.tar.gz
+          pip install -U "cplex" "qiskit-aer" "z3-solver" -c constraints.txt
+          mkdir -p /tmp/terra-tests
+          cp -r test /tmp/terra-tests/.
+          cp tools/verify_parallel_map.py /tmp/terra-tests/.
+          cp .stestr.conf /tmp/terra-tests/.
+          cp -r .stestr /tmp/terra-tests/. || :
+          sudo apt-get update
+          sudo apt-get install -y graphviz
+          pip check
+        displayName: 'Install dependencies'
```

Comment:
This might be something I overlooked - I didn't notice we did it differently between different runs (in part because it used to be in one monolithic file).  I'll update this in this PR to have two separate paths through, and do the test explicitly in one of them.

Reply:
I think it's probably fine to leave it. The sdist process is just a bit slower and more verbose but it shouldn't make a difference that we're only using the sdist path for linux.

Reply:
I added the option, since I was making changes, and made the qpy/image test run install from the sdist in 68a8ece9d.